lib/active_merchant/billing/gateways/shift4.rb

```diff
@@ -159,10 +175,12 @@ def add_card(post, credit_card, options)
           post[:card][:securityCode] = {}
           post[:card][:securityCode][:indicator] = 1
           post[:card][:securityCode][:value] = credit_card.verification_value
+          post[:card][:present] = 'Y'
```

Comment:
Two things:
1. We should default to `'N'` because an ecommerce transaction, even when it involves a credit card, is considered a "card not present" transaction.
2. From the gateway docs: `This should be set appropriately in the initial authorization/sale request. In subsequent requests, this field should be left blank or should not be sent.`

Therefore, I think we should remove this line as well as the line `post[:card][:present] = options[:present]` out of this method and into a method called `add_card_present`, which should only be called on authorize or purchase transactions.

Something like:
```
def add_card_present(post, options)
  post[:card] = {} unless post[:card].present?
 
  post[:card][:present] = options[:present] || 'N'
end
```